keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1504 @@
+  runtimes.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+* Extend the model that kube-scheduler has about
+  resources. Instead, it will need information from the plugin for
+  each resource request to determine where a Pod using the resource
+  might run. The [Representing Compute Resources in Kubernetes
+  proposal](https://docs.google.com/document/d/1666PPUs4Lz56TqKygcy6mXkNazde-vwA7q4e5H92sUc/edit#)
+  had some ideas what information the scheduler might need (“supports
+  overcommit”, “fractional”), but ultimately any choice regarding that
+  will only work for certain kinds of resources.
+
+* Standardize how to describe available resources. Only allocated
+  resources are visible through the APIs defined below. How to
+  describe available resources is plugin specific because it depends
+  on the kind of resource which attributes might be relevant. Plugins
+  should use and document their individual approach for this (for
+  example, defining a CRD and publishing through that).
+
+* Provide an abstraction layer for resource requests, i.e., something like a
+  “I want some kind of GPU”. Users will need to know about specific
+  resource plugins and which parameters they support. Portability of
+  workloads could be added on top of this proposal by introducing the
+  selection of a resource implementation through labels and
+  standardizing those labels and the associated parameters. The
+  [Resource Class
+  Proposal](https://docs.google.com/document/d/1qKiIVs9AMh2Ua5thhtvWqOqW0MSle_RV3lfriO1Aj6U/edit#heading=h.jzfmfdca34kj)
+  included such an approach.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+The proposal is that a plugin handles all operations that are specific
+to the resources managed by that plugin. This includes operations at
+the control plane level (tracking where in the cluster resources are
+available, helping with pod scheduling decisions, allocating resources
+when requested) as well as the node level (preparing container
+startup). Such a plugin can be implemented in arbitrary programming
+languages as long as it supports the resource allocation protocol and
+gRPC interfaces defined in this KEP. An utility package with Go
+support code will be made available to simplify the development of
+such a plugin, but using it will not be required and its API is not
+part of this KEP.
+
+Three new API object types get added in a new API group:
+- ResourcePlugin, not namespaced, with a description of the plugin.
+- ResourceClass, not namespaced, with privileged parameters for
```

Comment:
I named it after `StorageClass`, which is an API object in Kubernetes. `ResourceClass` serves a very similar purpose here:
- one level of indirection: the admin decides about plugin names during installation, the user only needs to know the class name
- it holds parameters that only an admin is allowed to set

The second point is the important one right now. The first one might become useful at some point.

I think consistency with the official API is more important than potential confusion with discussions that never made it into Kubernetes: many people will know StorageClass, while only a few heard about those other "ResourceClass" ideas.

Reply:
Fair enough, since ResourceClass never made it past discussion stage and we have StorageClass as an official API and this design is heavily inspired by that work, makes sense to stick to `ResourceClass` as the name.